pkg/arvo/mar/belt.hoon

```diff
@@ -0,0 +1,29 @@
+::  belt: runtime belt structure
+::
+|_  =belt:dill
+++  grad  %noun
+::  +grab: convert from
+::
+++  grab
+  |%
+  ++  noun  belt:dill
+  ++  json
+    ^-  $-(^json belt:dill)
+    =,  dejs:format
+    %-  of
+    :~  aro+(su (perk %d %l %r %u ~))
+        bac+ul
+        ctl+`$-(json @c)`so
+        del+ul
+        met+`$-(json @c)`so
+        ret+ul
+        txt+`$-(json (list @c))`sa
```

Comment:
It's kind of wild that these casts work. But the code is wrong, `so` should be `(cu taft so)`. As for `%txt`, you may want to send an array of strings instead of a single string -- this would allow more flexibility in calculating the cursor position in the presence of combining codepoints. In which case, `sa` would be `(ar (cu taft so))`.

Reply:
> you may want to send an array of strings instead of a single string -- this would allow more flexibility in calculating the cursor position in the presence of combining codepoints

Can you clarify a bit? Am I correct in reading "array of strings" as "array of strings that are, semantically, a single character"? I'm not sure what that would gain us over `(cu tuba sa)` here. Were you maybe thinking of doing the equivalent for _blits_ instead?

(I guess doing arrays for `%lin` blits (`(list @c)`) means we'd have to do them here, too, for consistency.)

Reply:
`@c` is equivalent to `@t`, but the %dill protocol uses `(list @c)` (ie, `$tour`). The JSON equivalent of that is most likely an array of (usually individual) character strings. The reason to do that (and, I think, the reason for the factoring of the %dill protocol) is that combining codepoints equivalent to a logically-singular "character" can be controlled by the client. Note that cursor positions are calculated based on list items ...

If all that seems like too much to tackle now, you'll at least want to do something about this current `%txt` parser, which will only work for ASCII.

Reply:
You are, once again, entirely right. Turns out cursor positioning was broken anyway. Fixed!

Reply:
`@c` is extremely not equivalent to `@t` what. `so:jo` produces `@t` not `@c`
`@t` is utf-8 encoded, `@c` is raw UTF-32 codepoints
```
> `@ux`~-~1f914.~21.~21.
0x21.0000.0021.0001.f914
> ? ~-1f914.~21.~21.
  @c
~-1f914.~21.~21.
> `@ux`(tuft ~-~1f914.~21.~21.)
0x2121.94a4.9ff0
> ? (tuft ~-~1f914.~21.~21.)
  @t
'🤔!!'
```

Reply:
_In the case of ascii_, you can just `;;` between `(list @cF)` and `(list @tD)`(`tape`) and everything will appear to work.

Reply:
@ohAitch, I worded that very poorly. What I meant is that a) `@c` is of arbitrary length, like `@t`, and b) the `(list @c)` conveys cursor positions/glyphs (ie, it's not like a tape).

I endorse the changes in eb40e2feed0bbd1a357398672d78d766777d308f.